### PR TITLE
sqflite integration

### PR DESCRIPTION
**Pull request by mikev-cw**
Removed sqflite_common_ffi dependency as apparently not used 
Renamed database class 
First attempt to implement all tables
All models has their CRUD methods
creation of demo bank accounts to show on home page

TODOs
Rem ffi dependency on tests too (they will not pass)

### CODE REVIEW

File: lib/model/budget.dart [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 
-const String budget = 'budget';
+const String budgetTable = 'budget';
 
 class BudgetFields extends BaseEntityFields {
+  static String id = 'id';
   static String name = 'name';
   static String amountLimit = 'amountLimit';
+  static String createdAt = 'createdAt';
+  static String updatedAt = 'updatedAt';
```
Comment (FedericoBruzzone) [inline lib/model/budget.dart:11]: The fields `id`, `createdAt` and `updateAt` are in the abstract class `BaseEntiryFileds`, why did you add them in this concrete class?

**Review by FedericoBruzzone — CHANGES_REQUESTED** [review]

**Comment by FedericoBruzzone** [comment]
After there changes, don't create a new commit. You can use `--amend` command.

**Comment by FedericoBruzzone** [comment]
would it be good to write some tests?

File: lib/model/budget.dart [hunk 2]
```diff
@@ -55,3 +59,64 @@ class Budget extends BaseEntity {
...
+      // reutrn null;
+    }
+  }
+
+  Future<List<Budget>> selectAll() async {
+    final database = await SossoldiDatabase.instance.database;
+
+    final orderByASC = '${BudgetFields.createdAt} ASC';
+
+    // final result = await database.rawQuery('SELECT * FROM $tableExample ORDER BY $orderByASC')
```
[the remark this answers is not in the document]
Comment (mikev-cw) [reply lib/model/budget.dart:94]: Not my own comments, they've been added by some previous commits.
I think they served to indicate an example of a free query, without using prepared statements.

**Comment by mikev-cw** [comment]
I need to add at least one other commit for fixing tests

File: lib/model/budget.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (FedericoBruzzone) [reply lib/model/budget.dart:93]: Can you remove this comments?

File: lib/model/budget.dart [hunk 3]
```diff
@@ -55,3 +59,64 @@ class Budget extends BaseEntity {
...
+
+  Future<int> updateItem(Budget item) async {
+    final database = await SossoldiDatabase.instance.database;
+
+    // You can use `rawUpdate` to write the query in SQL
+    return database.update(
+      budgetTable,
+      item.toJson(),
+      where:
+      '${BudgetFields.id} = ?', // Use `:` if you will not use `sqflite_common_ffi`
```
[the remark this answers is not in the document]
Comment (FedericoBruzzone) [reply lib/model/budget.dart:107]: Can you remove this comments?

File: lib/model/budget.dart (hunk 1 above)
Comment (mikev-cw) [reply lib/model/budget.dart:11]: `BaseEntityFields` is an _abstract_ class, so its properties MUST be declared in extended class, but they are not automatically inherited. If you remove line 7, you will get "_The getter 'id' isn't defined for the type 'BudgetFields'._" error on `_createDB()` func.

BTW there's a error anyway here, on lines 7, 10, and 11 that should be:
`  static String id = BaseEntityFields.getId;
  static String name = 'name';
  static String amountLimit = 'amountLimit';
  static String createdAt = BaseEntityFields.getCreatedAt;
  static String updatedAt = BaseEntityFields.getUpdatedAt;`
  for all models. I will edit they.

**Comment by mikev-cw** [comment]
So it seems FFI dependency is needed for tests. I restored pubspec.yaml and resolved all issues.

File: lib/model/budget.dart (hunk 1 above)
Comment (FedericoBruzzone) [reply lib/model/budget.dart]: Usually a field declared in the abstract class if it is public, the concrete class can use it without problems. This in practically any object oriented programming language. Maybe in dart it works differently...

**Comment by FedericoBruzzone** [comment]
Can I merge this PR right now?

**Comment by mikev-cw** [comment]
I'm ok with it. If you too, we can merge
